src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1684,6 +1670,9 @@ protected Verb verb()
             // we build this ONLY to perform the sufficiency check that happens on construction
             ReplicaPlans.forWrite(keyspace, cm.consistency(), tk, ReplicaPlans.writeAll);
 
+            // This host isn't a replica, so mark the request as being remote.
+            writeMetrics.remoteRequests.mark();
+
```

Comment:
If this host is a replica, `applyCounterMutationOnCoordinator()` -> `performWrite()`, and there we'll mark a local request against the metrics.

Reply:
Is it worth being explicit in this comment that the other path will ultimately update the `localRequests`